dev/ProjectReunion_BootstrapDLL/MddBootstrap.cpp

```diff
@@ -0,0 +1,141 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include "MddBootstrap.h"
+
+#include "IDynamicDependencyLifetimeManager.h"
+
+HRESULT GetFrameworkPackageInfoForPackage(PCWSTR packageFullName, const PACKAGE_INFO*& frameworkPackageInfo, wil::unique_cotaskmem_ptr<BYTE[]>& packageInfoBuffer);
+HRESULT AddFrameworkToPath(PCWSTR  path);
+
+IDynamicDependencyLifetimeManager* g_lifetimeManager = nullptr;
+DLL_DIRECTORY_COOKIE g_dllDirectoryCookie = 0;
+wil::unique_cotaskmem_string g_frameworkPath;
+
+STDAPI MddBootstrapInitialize(
+    const CLSID& appDynamicDependencyLifetimeManager)
+{
+    wil::com_ptr_nothrow<IDynamicDependencyLifetimeManager> lifetimeManager;
+    RETURN_IF_FAILED(CoCreateInstance(appDynamicDependencyLifetimeManager, nullptr, CLSCTX_LOCAL_SERVER, __uuidof(IDynamicDependencyLifetimeManager), reinterpret_cast<void**>(lifetimeManager.addressof())));
+
+    RETURN_IF_FAILED(lifetimeManager->Initialize());
+
+    wil::unique_cotaskmem_string packageFullName;
+    RETURN_IF_FAILED(lifetimeManager->GetPackageFullName(&packageFullName));
+
+    wil::unique_cotaskmem_ptr<BYTE[]> packageInfoBuffer;
+    const PACKAGE_INFO* frameworkPackageInfo = nullptr;
+    RETURN_IF_FAILED(GetFrameworkPackageInfoForPackage(packageFullName.get(), frameworkPackageInfo, packageInfoBuffer));
+
+    RETURN_IF_FAILED(AddFrameworkToPath(frameworkPackageInfo->path));
+
+    g_lifetimeManager = lifetimeManager.detach();
+    return S_OK;
+}
+
+STDAPI_(void) MddBootstrapShutdown()
+{
+    if (!g_lifetimeManager)
+    {
+        (void)LOG_IF_FAILED(g_lifetimeManager->Shutdown());
+        g_lifetimeManager->Release();
+        g_lifetimeManager = nullptr;
+    }
+
+    if (g_dllDirectoryCookie != 0)
+    {
+        RemoveDllDirectory(g_dllDirectoryCookie);
+        g_dllDirectoryCookie = 0;
+    }
+
+    if (g_frameworkPath)
+    {
+        wil::unique_cotaskmem_string path;
+        const auto hr = wil::TryGetEnvironmentVariableW(L"PATH", path);
+        if (SUCCEEDED(hr) && path)
+        {
+            const auto frameworkPathLength = wcslen(g_frameworkPath.get());
+            PCWSTR pathWithoutFrameworkPath = g_frameworkPath.get() + frameworkPathLength;
+            if (*pathWithoutFrameworkPath == L';')
+            {
+                ++pathWithoutFrameworkPath;
+            }
+            (void)LOG_IF_WIN32_BOOL_FALSE(SetEnvironmentVariableW(L"PATH", pathWithoutFrameworkPath));
+        }
+    }
+}
+
+HRESULT GetFrameworkPackageInfoForPackage(PCWSTR packageFullName, const PACKAGE_INFO*& frameworkPackageInfo, wil::unique_cotaskmem_ptr<BYTE[]>& packageInfoBuffer)
+{
+    frameworkPackageInfo = nullptr;
+
+    wil::unique_package_info_reference packageInfoReference;
```

Comment:
This code's a little light on comments explaining what's happening, and there's multiple independent-ish blocks. Either factor into methods that have clear purposes, or put a little one-liner explaining the purpose of the block.